daemon/list.go

```diff
@@ -162,10 +162,7 @@ func (daemon *Daemon) filterByNameIDMatches(view container.View, ctx *listContex
 	cntrs := make([]container.Snapshot, 0, len(matches))
 	for id := range matches {
 		c, err := view.Get(id)
-		if err != nil {
-			return nil, err
-		}
-		if c != nil {
+		if err == nil {
```

Comment:
I agree with @thaJeztah it'd be better to check for `err != nil` and error out instead of silencing the error. Rationale: when someone adds a new error type that view.Get could return (i know it's unlikely but still), they wouldn't have to update all callers.